docs/source/complex_numbers.rst

```diff
@@ -0,0 +1,155 @@
+.. _complex_numbers-doc:
+
+Complex Numbers
+===============
+
+Complex Numbers are numbers that can be expressed in the form :math:`a + bi`, where a and b are real numbers,
+and *i* is a solution of the equation :math:`x^2 = −1`. Complex numbers frequently occur in mathematics and
+engineering, especially in signal processing. The aim of introducing Complex Tensors is to provide a more natural
+user experience for users and libraries (eg. TorchAudio) that currently workaround the lack of Complex Tensors
+by using Float Tensors with shape :math:`(..., 2)` where the last dimension contains the real and imaginary values.
+
+We support many functions for Complex Tensors eg. SVD, QR etc. Operations on complex tensors are likely to be
+faster than operations on float tensors mimicking them. We also have a `Vec256 class` for `ComplexFloat` and
+`ComplexDouble` to benefit from Intel assembly instructions for performing vectorized operations on CPU. In
+addition, operations on Complex Tensors utilize specialized kernels for complex provided by LAPACK (for CPU)
+and BLAS (for CUDA) or kernels in `torch` specially written for complex tensors. Thus using functions Complex
+Tensors will provide performance benefits as opposed to users defining their own functions.
+
+.. warning ::
+     Complex Tensors is a beta feature and subject to change.
+
+Creating Complex Tensors
+----------------------
+
+We support two complex dtypes: `torch.cfloat` and `torch.cdouble`
+
+::
+
+     >>> x = torch.randn(2,2, dtype=torch.cfloat)
+     >>> x
+     tensor([[-0.4621-0.0303j, -0.2438-0.5874j],
+          [ 0.7706+0.1421j,  1.2110+0.1918j]])
+
+The default dtype for complex tensors is updated based on the current default tensor type.
+cdouble if double else float.
+
+The following factory functions can be used to create complex tensors:
+
+- :func:`torch.tensor`
+- :func:`torch.as_tensor`
+- :func:`torch.empty`
+- :func:`torch.rand`
+- :func:`torch.randn`
+- :func:`torch.as_strided`
+- :func:`torch.from_numpy`
+- :func:`torch.zeros`
+- :func:`torch.ones`
+- :func:`torch.full`
+- :func:`torch.eye`
+
+Transition from the old representation
+--------------------------------------
+
+Users who currently worked around the lack of complex tensors with real tensors of shape `(..., 2)`
+can easily to switch using the complex tensors in their code using :func:`torch.view_as_complex` and
+- :func:`torch.view_as_real`:
```

Comment:
I'd suggest mentioning that this involves ZERO copying

Reply:
done!